src/nix/show-config.cc

```diff
@@ -9,15 +9,44 @@ using namespace nix;
 
 struct CmdShowConfig : Command, MixJSON
 {
+    std::optional<std::string> name;
+
+    CmdShowConfig() {
+        expectArgs({
+            .label = {"name"},
+            .optional = true,
+            .handler = {&name},
+        });
+    }
+
     std::string description() override
     {
-        return "show the Nix configuration";
+        return "show the Nix configuration or the value of a specific setting";
     }
 
     Category category() override { return catUtility; }
 
     void run() override
     {
+        if (name) {
+            if (json) {
+                throw UsageError("'--json' is not supported when specifying a setting name");
```

Comment:
Shouldn't we allow that? Both for consistency and because some settings are themselves a structured value (`substituters` for example) so it's worth having them already parsed.

Reply:
Ah, I apparently didn't see the second half of the PR description :man_facepalming: 

Wouldn't it be enough to just print `setting->second.toJSON()` in [here](https://github.com/NixOS/nix/blob/1fc74afbbae1ae3e78e3c0d2096e81c1e24f9d52/src/nix/show-config.cc#L43-L44) ?

Reply:
Sadly, `SettingInfo` doesn't have `toJSON()`... Which is why I left it unsupported :sweat_smile: 

Also, `SettingInfo` is stringly-typed, so I'm not sure if we can easily get (useful) structured values out of it?

Reply:
Though, if `configRegistrations` was somehow exposed, maybe we could go through that, since that appears to have `toJSON` _with_ proper JSON types (that's how `globalConfig.toJSON()` is implemented from what I can see).

Reply:
Derp, it is exposed, just under `globalConfig`. Still, I did a bit of playing and it's real ugly, trying to get the proper `toJSON` stuff. Basically need to be able to do the following:

1. Iterate through all of the `Config`s in `configRegistrations`
1. Gain access to `Config::_settings` (which is `private`) so we can look up the setting by key there
2. Gain access to `AbstractSetting::toJSON()` (which is `protected`)

If there was a general-purpose "C++ value to appropriate JSON value" function somewhere, we could just call that, but I don't know if that exists (all the json files appears to be under libexpr).

Reply:
I don't think we need JSON support here, because you can always get a specific flag by doing `nix show-config --json | jq .<name>.value`.

Reply:
I might follow an issue. Whether or not this an important feature (and I readily believe the `jq` way is good enough), it would be nice if our configuration infrastructure didn't get in the way of this.

Reply:
I'd be happy to see json supported, but my attempt was a bit of a pain, and I was in over my head.